src/utils/account/refreshAccount.tsx

```diff
@@ -12,10 +12,9 @@ const setNewAccount = async () => {
       const account = await getAccount(address);
       if (account != null) {
         const accountData = {
-          balance: account.balance,
+          ...account,
           address,
-          nonce: getLatestNonce(account),
-          username: account.username
+          nonce: getLatestNonce(account)
         };
         store.dispatch(setAccount(accountData));
```

Comment:
maybe we should create a redux action "updateAccountData" and just dispatch the values that need to be updated and to the merging inside redux

Reply:
account comes from API. It's OK to only update the nonce